challenge-react/src/App.js

```diff
@@ -1,111 +1,111 @@
+  text-align: center;
+`;
+
+const CardContainer = styled.div`
+  display: flex;
+
+  @media (max-width: 1000px) {
+    flex-direction: column;
+    padding: 2.5% 20%;
+
+    > * {
+      flex: 0 0 80%;
+      margin: 5%;
     }
+  }
 
-    render() {
-      const self = this;
-      const cards = this.state.charities.map(function (item, i) {
-        const payments = [10, 20, 50, 100, 500].map((amount, j) => (
-          <label key={j}>
-            <input
-              type="radio"
-              name="payment"
-              onClick={function () {
-                self.setState({ selectedAmount: amount });
-              }}
-            />
-            {amount}
-          </label>
-        ));
-
-        return (
-          <Card key={i}>
-            <p>{item.name}</p>
-            {payments}
-            <button
-              onClick={handlePay.call(
-                self,
-                item.id,
-                self.state.selectedAmount,
-                item.currency
-              )}
-            >
-              Pay
-            </button>
-          </Card>
-        );
-      });
+  @media (min-width: 1001px) {
+    flex-wrap: wrap;
+    margin: auto;
+    max-width: 1000px;
+    position: relative;
 
-      const style = {
-        color: 'red',
-        margin: '1em 0',
-        fontWeight: 'bold',
-        fontSize: '16px',
-        textAlign: 'center',
-      };
-
-      const donate = this.props.donate;
-      const message = this.props.message;
-
-      return (
-        <div>
-          <h1>Tamboon React</h1>
-          <p>All donations: {donate}</p>
-          <p style={style}>{message}</p>
-          {cards}
-        </div>
-      );
+    > * {
+      flex: 0 0 45%;
+      margin: 2.5%;
     }
   }
-);
-
-/**
- * Handle pay button
- * 
- * @param {*} The charities Id
- * @param {*} amount The amount was selected
- * @param {*} currency The currency
- * 
- * @example
- * fetch('http://localhost:3001/payments', {
+`;
+
+const API_URL = 'http://localhost:3001';
+
+const App = () => {
+  const [charities, setCharities] = useState([]);
+  const [currentDisplayOverlayId, setCurrentDisplayOverlayId] = useState();
+  const [totalDonation, setTotalDonation] = useState(0);
+  const [infoMessage, setInfoMessage] = useState();
+
+  useEffect(() => {
+    if (infoMessage) {
+      const timer = setTimeout(() => {
+        setInfoMessage(undefined);
+      }, 5000);
+      return () => clearTimeout(timer);
+    }
+  }, [infoMessage]);
+
+  useEffect(() => {
+    fetchCharities();
+    fetchPayments();
+  }, []);
+
+  const fetchCharities = () => {
+    fetch(`${API_URL}/charities`)
+      .then((resp) => resp.json())
+      .then((charities) => setCharities(charities));
+  };
+
+  const fetchPayments = () => {
+    fetch(`${API_URL}/payments`)
+      .then((resp) => resp.json())
+      .then((payments) => {
+        const totalDonation = payments.reduce(
+          (totalDonation, payment) => (totalDonation += payment.amount),
```

Comment:
![Screen Shot 2021-01-25 at 10 36 00](https://user-images.githubusercontent.com/10527554/105658495-237a0900-5ef9-11eb-8021-79e390051d9a.png)

It seems you forgot to install eslint